src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -44,24 +50,58 @@ public TextSubmission save(TextSubmission textSubmission, TextExercise textExerc
         // update submission properties
         textSubmission.setSubmissionDate(ZonedDateTime.now());
         textSubmission.setType(SubmissionType.MANUAL);
-        textSubmission.setParticipation(participation);
+
+        boolean isExampleSubmission = textSubmission.isExampleSubmission() != null && textSubmission.isExampleSubmission();
+
+        // Example submissions do not have a participation
+        if (!isExampleSubmission) {
+            textSubmission.setParticipation(participation);
+        }
         textSubmission = textSubmissionRepository.save(textSubmission);
 
-        participation.addSubmissions(textSubmission);
+        // All the following are useful only for real submissions
+        if (!isExampleSubmission) {
+            participation.addSubmissions(textSubmission);
 
-        User user = participation.getStudent();
+            User user = participation.getStudent();
 
-        if (textSubmission.isSubmitted()) {
-            participation.setInitializationState(InitializationState.FINISHED);
-        } else if (textExercise.getDueDate() != null && !textExercise.isEnded()) {
-            // save submission to HashMap if exercise not ended yet
-            AutomaticSubmissionService.updateSubmission(textExercise.getId(), user.getLogin(), textSubmission);
-        }
-        Participation savedParticipation = participationRepository.save(participation);
-        if (textSubmission.getId() == null) {
-            textSubmission = savedParticipation.findLatestTextSubmission();
+            if (textSubmission.isSubmitted()) {
+                participation.setInitializationState(InitializationState.FINISHED);
+            } else if (textExercise.getDueDate() != null && !textExercise.isEnded()) {
+                // save submission to HashMap if exercise not ended yet
+                AutomaticSubmissionService.updateSubmission(textExercise.getId(), user.getLogin(), textSubmission);
+            }
+            Participation savedParticipation = participationRepository.save(participation);
+            if (textSubmission.getId() == null) {
+                textSubmission = savedParticipation.findLatestTextSubmission();
+            }
         }
 
         return textSubmission;
     }
+
+    // Find a text submission without assessments
```

Comment:
Please use proper JavaDoc comments and describe in more detail what you are trying to achieve. E.g. what does "findAny()" mean here? Is it random or is it the first one that fulfills the criteria?
Also I assume that this code executes many database requests. Could we not write a database query in the corresponding repository to optimize this and execute one query in the database?

Reply:
To be honest I am not sure how to write such query: we need to retrieve a random participation, with at  least a submitted text submission, where the latest submitted text submission has no result associated